src/rcms/fm/app/level1/HCALlevelTwoEventHandler.java

```diff
@@ -185,89 +253,15 @@ public void initAction(Object obj) throws UserActionException {
         thread2.start();
       } 
 
-      // start the TriggerAdapter watchdog thread
-      System.out.println("[HCAL LVL2 " + functionManager.FMname + "] Starting TriggerAdapter watchdog thread ...");
-      logger.debug("[HCAL LVL2 " + functionManager.FMname + "] StartingTriggerAdapter watchdog thread ...");
-      TriggerAdapterWatchThread thread3 = new TriggerAdapterWatchThread();
-      thread3.start();
-      functionManager.parameterSender.start();
-
-
-      // check run type passed from Level-1
-      if(((StringT)parameterSet.get("HCAL_RUN_TYPE").getValue()).getString().equals("local")) {
-
-        RunType = "local";
-
-        // request a session ID
-        //getSessionId();
-        // Get SID from LV1:
-        if (parameterSet.get("SID") != null) {
-          Sid = ((IntegerT)parameterSet.get("SID").getValue()).getInteger();
-          functionManager.getParameterSet().put(new FunctionManagerParameter<IntegerT>("SID",new IntegerT(Sid)));
-          functionManager.getParameterSet().put(new FunctionManagerParameter<IntegerT>("INITIALIZED_WITH_SID",new IntegerT(Sid)));
-          logger.info("[Martin log HCAL LVL2 " + functionManager.FMname + "] Received the following SID from LV1 :"+ Sid) ;
-        }
-        else {
-          String warnMessage = "[Martin log HCAL LVL2 " + functionManager.FMname + "] Did not receive a SID from LV1...";
-          logger.warn(warnMessage);
-        }
-
-
-        GlobalConfKey = "not used";
-
-        // set the run type in the function manager parameters
-        functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>("HCAL_RUN_TYPE",new StringT(RunType)));
-        functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>("GLOBAL_CONF_KEY",new StringT(GlobalConfKey)));
+      // start the TriggerAdapter watchdog thread. Note: containerTriggerAdapter is filled after this. Start watchThread with role check.
+      if (functionManager.FMrole.equals("EvmTrig")){
```

Comment:
only EvmTrig FM Starts watchthread, didn't use `containerTA` because it's not filled yet.